Formula/pyenv.rb

```diff
@@ -9,7 +9,7 @@ class Pyenv < Formula
 
   livecheck do
     url :stable
-    strategy :github_latest
+    regex(/^v?(\d+(?:\.\d+)+(-\d+)?)$/i)
```

Comment:
```suggestion
    regex(/^v?(\d+(?:[.-]\d+)+)$/i)
```

In some situations, it's better to match tags like `1.2.3-4` by substituting `[.-]` as the delimiter in the standard regex. In this case, there's one older tag that uses a format like `v0.4.0-20140110.1` instead of the typical `v0.4.0-20140110` format at the time. The PR regex wouldn't match it but this suggestion will.

Reply:
Actually, it seems to me that we _don't_ want to match that tag format, since it comes from a scheme they'd long abandoned.

Reply:
It's not that we want to specifically match the old `v0.4.0-20140110.1` tag but rather just to demonstrate that only accounting for an optional `-\d+` suffix may lead to missed versions. Projects always use a predictable tag format until they don't and this one has used a few different variations over time, so I think the flexibility is preferable here.

Reply:
I could sink time into gathering data to answer that question (not until later today at the earliest) but I don't think the answer would really impact this decision. Even if the answer is no, it's fully possible that the available examples may have simply not encountered a different version *yet* and that doesn't say much.

With the regex options here, the explicit approach (PR regex) can potentially miss versions and the looser approach can potentially match versions we don't want. It's hard to identify a missed version (we have to do this outside of livecheck) but it's much easier to identify an inappropriate version and update the `livecheck` block. When faced with these options with respect to this particular context, the relatively looser option would be preferable in my experience.

Reply:
> Even if the answer is no, it's fully possible that the available examples may have simply not encountered a different version _yet_ and that doesn't say much.

I think that argument cuts both ways. That is, in this case, I would say that it likely doesn't say much that we've used the approach proposed in this suggestion in a number of other formulae and not encountered problems either. 

> It's hard to identify a missed version (we have to do this outside of livecheck) but it's much easier to identify an inappropriate version and update the _livecheck_ block.

I think this depends on the formula. For formulae with few users, I agree with you that it's hard to identify a missed version. For formulae as popular as this one (nearly 130k installs in a month...), I think we'd find out about missed versions relatively quickly because a user will update the formula. I know this because that's how I discovered the `livecheck` block was broken: #93925

In cases like this it seems much more prudent to choose stricter matches to avoid mistaken version bumps that are mistakenly approved (and we've both seen a few of those already). If that comes at the cost of missing some versions: I'm fine with that. Missing versions can be fixed: just update the formula. Mistaken version bumps are much harder to take back: #92523

Reply:
> I think that argument cuts both ways. That is, if that is indeed the case, I would say that it likely doesn't say much that we've used the approach proposed in this suggestion in a number of other formulae and not encountered problems either.

I don't disagree. It comes down to whether we would prefer to miss a version (requiring an external intervention) or to potentially report an inappropriate version and this really depends on the context of a given formula.

> I think this depends on the formula. For formulae with few users, I agree with you that it's hard to identify a missed version. For formulae as popular as this one (nearly 130k installs in a month...), I think we'd find out about missed versions relatively quickly because a user will update the formula.

When framed this way, I can agree with that idea. I still feel that the looser approach may be preferable in general (i.e., better to detect a potentially unwanted version rather than miss a version) but I accept that the potential detriments of the more explicit approach aren't as much of a concern for a very popular formula like this. If the more explicit regex ends up regularly missing versions in the future, we can always modify it in the future.